### DISCUSSION TITLE
Using "access" and "fail-on" with thumbnail_buffer

### DISCUSSION THREAD

**Discussion by kstanikviacbs** [libvips/libvips#2781] [Q&A] [answered]
Hi! 

I attempted to pass "access" and "fail-on" params to ```vips_thumbnail_buffer```, but with no luck and getting errors like:
```
thumbnail_buffer: no property named `fail-on'
```
This made me thinking that probably due to thumbnail_buffer processing nature it does not make sense actually to pass them at all and there's the reason of the error - they are simply not supported. 

And if I want to fail on truncated image I need to slow down processing a bit instead by firstly loading the image to buffer with ```vips_image_new_from_buffer``` and fail-on passed and calling thumbnail then if the first one is successful. 

Is my way of understanding it correct?

**Comment by jcupitt** [comment, marked as answer]
Hi @kstanikviacbs,

You're right, the `thumbnail` operators don't support fail-on. Perhaps they should? It would be a simple thing to add.

You could do `new_from_file` and then thumbnail that, but it'd be a LOT slower.

**Reply by jcupitt** [reply]
Btw, you can check what arguments an operator supports with eg,:

```
$ vips thumbnail
generate thumbnail from file
usage:
   thumbnail filename out width [--option-name option-value ...]
where:
   filename     - Filename to read from, input gchararray
   out          - Output image, output VipsImage
   width        - Size to this width, input gint
			default: 1
			min: 1, max: 10000000
optional arguments:
   height       - Size to this height, input gint
			default: 1
			min: 1, max: 10000000
   size         - Only upsize, only downsize, or both, input VipsSize
			default: both
			allowed: both, up, down, force
   no-rotate    - Don't use orientation tags to rotate image upright, input gboolean
			default: false
   crop         - Reduce to fill target rectangle, then crop, input VipsInteresting
			default: none
			allowed: none, centre, entropy, attention, low, high, all
   linear       - Reduce in linear light, input gboolean
			default: false
   import-profile - Fallback import profile, input gchararray
   export-profile - Fallback export profile, input gchararray
   intent       - Rendering intent, input VipsIntent
			default: relative
			allowed: perceptual, relative, saturation, absolute
operation flags: nocache 
```

**Reply by kstanikviacbs** [reply]
Understood, thanks for explanation @jcupitt . So if it comes to me as a vips user - introducing ```fail-on``` into thumbnail operators would be really beneficial - I would be able then to control the failing behaviour while still taking advantage of thumbnail time of execution. 

P.S Thanks for the tip for checking supported params - that's really useful.

**Reply by kstanikviacbs** [reply]
I'd be actually happy to contribute to this change if not the main project I'm involved in right now which takes a lot of my time.

**Reply by jcupitt** [reply]
I added this feature to 8.13:

https://github.com/libvips/libvips/commit/f06c9f3bcffec2c9f09c8ca460ac534cc03c2188

Thanks for the suggestion!

**Reply by jcupitt** [reply]
... you can use it like this:

```
$ vips thumbnail truncated.jpg x.jpg 100 --fail-on truncated
...
$ echo $?
1
$ vips thumbnail truncated.jpg x.jpg 100 
...
$ echo $?
0
```

**Reply by kstanikviacbs** [reply]
Thanks a lot for so quickly addressing it!

**Comment by kstanikviacbs** [comment]
@jcupitt Hi again! Refreshing the topic as I've realised that since 8.13.x this one fails ```vips thumbnail corrupted.jpg test.jpg 200``` for partially corrupted image and vips warnings, which was not the case before (8.12.x). Is there any chance you can double check it on your side? BTW - passing different ```fail-on``` options have no effect as well.

**Comment by kstanikviacbs** [comment]
Created an issue: https://github.com/libvips/libvips/issues/2973.
